bsh-bsf-engine/src/main/java/bsh/util/BeanShellBSFEngine.java

```diff
@@ -52,7 +52,9 @@ public void initialize ( BSFManager mgr, String lang, Vector declaredBeans)
         try {
             interpreter.set( "bsf", mgr );
         } catch ( EvalError e ) {
-            throw new BSFException(BSFException.REASON_OTHER_ERROR, "bsh internal error: "+e, e);
+        	BSFException bsf = new BSFException(BSFException.REASON_OTHER_ERROR, "bsh internal error: "+e, e);
+        	bsf.initCause(e);
```

Comment:
This is redundant as we are already passing `e` to the constructor.

Reply:
Happy to remove this, however the BSFException(String , Throwable) does not chain exceptions.  AFAIK the BSFException class constructor:

 public BSFException (int reason, String msg, Throwable t) {
	this (reason, msg);
	**targetThrowable = t;**
  }

targetThrowable isn't chained, and when the exception is thrown the only way to get they caused by is to use the method:

**public Throwable getTargetException ()**

This is the reason why I've initCaused(throwable) so the exception becomes chained.

Reply:
Happy to make this change, although the reason why this done is to ensure the BSFException supported Exception chaining.  

AFAIK The BSFException class does not populate the cause of the exception use the constructor:

public BSFException (int reason, String msg, **Throwable t**) {
    this (reason, msg);
    **targetThrowable = t;**
  }

Instead it tracks the throwable via a class variable which is exposed using the getTargetException() method.

Initialising the cause ensures they getCause() returns the expected, removing the initCause() stopped exception chaining working for me.

Reply:
@nickl- I'll hold off making the change until you've commented on the exception chaining issue and if the desire is to keep exception chaining disabled.  BR.

Reply:
I see well spotted!! `initCause` will be the desired route then.